micrometer-core/src/main/java/io/micrometer/core/instrument/push/PushRegistryConfig.java

```diff
@@ -90,6 +90,21 @@ default int batchSize() {
         return getInteger(this, "batchSize").orElse(10000);
     }
 
+    /**
+     * This controls when the call to {@link PushMeterRegistry#publish()} is scheduled. If
+     * this returns true, publishing is scheduled at the beginning of each step interval,
+     * relative to the Unix Epoch. This has the effect of causing all application
+     * instances publishing metrics with this configuration to publish at the same time,
+     * which may overload resources by concentrating the work to publish and ingest
+     * metrics. The more instances you have publishing metrics at the same time, the more
+     * of a problem this will be.
+     * @return false if publishing should be scheduled relative to registry instantiation
+     * time. Default is {@code false} to avoid the documented resource exhaustion issue.
+     */
+    default boolean isPushAlignedGlobally() {
```

Comment:
We could probably spend endless time coming up with better and better names, but here's a third iteration where I've avoided using "epoch" in the name because I think it will be less clear what this does compared to focusing on the effect of pushes being aligned globally if you set this to `true`. Anyone who has feedback about the naming, feel free to share (before we release it, ideally).

I've updated the JavaDoc to explain the issue that motivated this configuration being introduced and the reason for its default.